### PR TITLE
Add json option for manifest file creation

### PR DESCRIPTION
**Pull request by mmitche**
- Add a global option to darc to switch to yaml/json.
- Implement option for gather-drop's manifest generation

Uses newtonsoft.json rather than System.Text.Json because of the netstandard2.1 target

### CODE REVIEW

File: src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs [hunk 1]
```diff
@@ -383,11 +383,32 @@ private async Task WriteDropManifest(List<DownloadedBuild> downloadedBuilds, str
...
+            }
+        }
+
+        /// <summary>
+        ///     Write the manifest in yaml format
+        /// </summary>
+        /// <param name="downloadedBuilds">List of downloaded builds in the manifest</param>
+        /// <param name="outputPath">Output file path of the manifest</param>
+        /// <returns>Async task</returns>
+        private async Task WriteYamlDropManifest(List<DownloadedBuild> downloadedBuilds, string outputPath)
```
Comment (riarenas) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: [nit] add async suffix to both methods?

Comment (RussKie) [inline src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs]: [nit] async methods should have `Async` suffix to help in comprehension of the API

File: src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+namespace Microsoft.DotNet.Darc.Options
+{
+    /// <summary>
+    /// Output type that darc should use. Commandline enum parsing is case sensitive
+    /// so we put these as lower case.
+    /// </summary>
+    enum DarcOutputType
+    {
+        yaml,
```
Comment (RussKie) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs]:
```suggestion
        Yaml,
```

File: src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+        json
```
Comment (RussKie) [inline src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs]:
```suggestion
        Json
```

**Comment by mmitche** [comment]
> WriteYamlDropManifestAsync

Will change in next PR. More coming.

File: src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs (hunk 2 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs]: The command line parsing is case-sensitive, and the rest of our parameters are all lower case. I felt that having to specify "Json" is inconsistent.

File: src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs (hunk 3 above)
Comment (mmitche) [reply src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs]: The command line parsing is case-sensitive, and the rest of our parameters are all lower case. I felt that having to specify "Json" is inconsistent.

Comment (RussKie) [reply src/Microsoft.DotNet.Darc/src/Darc/Options/DarcOutputType.cs]: No worries

**Pull request merged**